### ISSUE TITLE
mangoplot not reading .csv files from mangohud

### ISSUE THREAD

**Issue by Kagukara** [flightlessmango/MangoHud#1362]
**Describe the bug**
When trying to use `mangoplot` with the logging files produced from `mangohud`, `mangoplot` can't read them.
```
$ mangoplot ~/.config/MangoHud/mangologs/
Loading 2 benchmark files
The folder 
   /home/richard/.config/MangoHud/mangologs 
contains no CSV file (make sure they have the .csv extension)
```
```
$ tree ~/.config/MangoHud/mangologs/
/home/richard/.config/MangoHud/mangologs/
├── vkcube_2024-06-23_19-10-50.csv
└── vkcube_2024-06-23_19-10-50_summary.csv

1 directory, 2 files
```
**List relevant hardware/software information**
- Linux Distribution: Arch Linux
- MangoHud version: v0.7.2-20-g2a2cc4a
- GPU: AMD Radeon RX 7900 XTX 

**To Reproduce**
Steps to reproduce the behaviour:
1. Open the terminal
2. Type `mangoplot ~/path/to/csv/folder`
   - Replacing `~/path/to/csv/files` with location to your folder with the .csv files.

**Expected behaviour**
For `mangoplot` to read the .csv files `mangohud` produces from logging.

**Comment by flightlessmango** [comment]
ping @AdelKS

**Comment by AdelKS** [comment]
Hey can you drag and drop the file `vkcube_2024-06-23_19-10-50.csv` ? Append `.txt` to it so GitHub accepts it : `vkcube_2024-06-23_19-10-50.csv.txt`

**Comment by Kagukara** [comment]
[vkcube_2024-06-23_19-10-50.txt](https://github.com/user-attachments/files/15946129/vkcube_2024-06-23_19-10-50.txt)

**Comment by AdelKS** [comment]
@flightlessmango it is just that there are extra "header" rows in the `csv` files now, which moved the title row that starts with `fps` (we look for that one to know when the actual benchmark data starts) past the 4th row,  we can look for the `fps` line in the first `100` rows instead of the first `4` and that fixes this issue
```patch
diff --git a/bin/mangoplot.py b/bin/mangoplot.py
index 7d82b3a..41d81f2 100755
--- a/bin/mangoplot.py
+++ b/bin/mangoplot.py
@@ -195,8 +195,9 @@ class BenchmarkFile:
 
             found_fps_column = False
             for row_number, row_content in enumerate(reader):
-                if row_number > 4:
-                    # if we're past the 4th row, break the loop
+                if row_number > 100:
+                    # did not find the row that starts with the
+                    # 'fps' column up until here.  give up.
                     break
```

**Comment by flightlessmango** [comment]
@AdelKS This issue stems from enabling the option `log_versioning`.
It has more information than the usual one, but it's not fully supported every where, like in mangoplot and on flightlessmango.com.
I would recommend we support both of them instead

**Comment by AdelKS** [comment]
Hey @flightlessmango yeah the patch I attached will make the code work with any mangohud CSV file who has the row that starts with fps anywhere in the first 100 rows (and then assumes every row after that to be actual data)

**Comment by Kagukara** [comment]
The patch above works for me.

**Comment by flightlessmango** [comment]
@AdelKS thanks!
It's fixed here 6c665653ba28e9296e0f89d07faceeb859f52872
